whois-commons/src/main/java/net/ripe/db/whois/common/rpki/RpkiService.java

```diff
@@ -20,31 +21,43 @@ public class RpkiService {
 
     private static final Logger LOGGER = LoggerFactory.getLogger(RpkiService.class);
 
-    private final NestedIntervalMap<Ipv4Resource, Set<Roa>> ipv4Tree = new NestedIntervalMap<>();
-    private final NestedIntervalMap<Ipv6Resource, Set<Roa>> ipv6Tree = new NestedIntervalMap<>();
+    private NestedIntervalMap<Ipv4Resource, Set<Roa>> ipv4Tree;
+    private NestedIntervalMap<Ipv6Resource, Set<Roa>> ipv6Tree;
+
+    private final RpkiDataProvider rpkiDataProvider;
 
     public RpkiService(final RpkiDataProvider rpkiDataProvider) {
+        this.rpkiDataProvider = rpkiDataProvider;
+        loadRoas();
+    }
+
+    @Scheduled(cron = "0 */15 * * * *")
+    private void loadRoas() {
         final List<Roa> loadedRoas = rpkiDataProvider.loadRoas();
         if (loadedRoas != null && !loadedRoas.isEmpty()){
             final List<Roa> roas = loadedRoas.stream()
                     .filter(roa -> roa.getTrustAnchor() != TrustAnchor.UNSUPPORTED)
                     .collect(Collectors.toList());
 
             LOGGER.info("downloaded {} roas from rpki", roas.size());
-            buildTrees(roas, ipv4Tree, ipv6Tree);
+
+            final Pair<NestedIntervalMap<Ipv4Resource, Set<Roa>>, NestedIntervalMap<Ipv6Resource, Set<Roa>>> trees = buildTree(roas);
```

Comment:
No need to return anything from `buildTrees` if you want to update *both* fields, do it directly in that method?

Reply:
yes, you are right. I don't need to return a pair.
I did update this